markdown/ch7.markdown

```diff
@@ -342,12 +342,12 @@ def align(reference, target, cost=cost_mse):
         res = optimize.minimize(cost, p, args=(ref, tgt), method='Powell')
         p = res.x
 
-        print('Pyramid level %i' % n)
-        print('Angle:', np.rad2deg(res.x[0]))
-        print('Offset:', res.x[1:] * 2 ** n)
-        print('Cost function:', res.fun)
-        print('')
+        # print current level, overwriting each time (like a progress bar)
+        print(f'Level: {n}, Angle: {np.rad2deg(res.x[0]) :.3}, '
+              f'Offset: ({res.x[1] * 2**n :.3}, {res.x[2] * 2**n :.3}), '
+              f'Cost: {res.fun :.3}', end='\r')
```

Comment:
@stefanv where? You mean before the colon? I thought it best to separate the formatting `:.3` from the expression being evaluated.